### PR TITLE
Doc - Add How to match a shape with optional fields?

### PR DESCRIPTION
**Pull request by Vadorequest**
Add "How to match a shape with optional fields?" section in readme.

Following discussion in https://github.com/Dean177/jest-to-match-shape-of/issues/75

### CODE REVIEW

**Comment by Dean177** [comment]
Amazing, thanks @Vadorequest

File: Readme.md [hunk 1]
```diff
@@ -102,6 +102,22 @@ describe('an api', () => {
...
 ```
 
+### How to match a shape with optional fields?
+
+Sometimes, the expected shape may vary during integration tests. (e.g: A field may be missing)
+
+If you want to make a shape allow optional fields, the simplest way is to remove those fields from the expected shape, as follow:
+
+```ts
+toMatchOneIf([{ ant: 17 }]) // bat is optional here
```
Comment (Dean177) [inline Readme.md:112]:
```suggestion
toMatchOneOf([{ ant: 17 }]) // bat is optional here
```

File: Readme.md [hunk 2]
```diff
@@ -102,6 +102,22 @@ describe('an api', () => {
...
+```
+
+A more robust alternative is to define all possible shapes, this way you still test the types of the properties: 
+
+```ts
+toMatchOneIf([{ ant: 17, bat: 176 }, { ant: 17 }]) // bat is still optional, but must be numeric
```
Comment (Dean177) [inline Readme.md:118]:
```suggestion
toMatchOneOf([{ ant: 17, bat: 176 }, { ant: 17 }]) // bat is still optional, but must be numeric
```

**Comment by Vadorequest** [comment]
Good catch, I had blindly copy/pasted yours from https://github.com/Dean177/jest-to-match-shape-of/issues/75#issuecomment-638743623 😅

**Pull request merged**